paasta_tools/deployd/common.py

```diff
@@ -50,7 +80,28 @@ def put(self, item, *args, **kwargs):
         Queue.put(self, item, *args, **kwargs)
 
 
-def rate_limit_instances(instances, number_per_minute, watcher_name):
+class PaastaPriorityQueue(PriorityQueue):
+
+    def __init__(self, name, *args, **kwargs):
+        self.name = name
+        PriorityQueue.__init__(self, *args, **kwargs)
+        self.counter = 0
```

Comment:
The counter is to preserve the FIFO nature of the queue. I'll add a comment because it isn't intuitive actually. Essentially the `PriorityQueue` does a sort on the first item in the tuple and then on the second etc. So the we sort first by priority and then by this counter (which always  increments).